app/components/channel_drawer/channels_list/list/list.js

```diff
@@ -40,29 +38,41 @@ class List extends PureComponent {
     constructor(props) {
         super(props);
 
+        this.state = {
+            sections: this.buildSections(props),
+            showIndicator: false,
+            width: 0
+        };
+
         MaterialIcon.getImageSource('close', 20, this.props.theme.sidebarHeaderTextColor).then((source) => {
             this.closeButton = source;
         });
     }
 
-    componentDidUpdate(prevProps) {
-        if (prevProps.unreadChannelIds !== this.props.unreadChannelIds && this.refs.list) {
+    componentWillReceiveProps(nextProps) {
+        if (nextProps !== this.props) {
```

Comment:
I'm pretty sure `nextProps !== this.props` is always true. You could do a shallowEquals on them or check the fields used by buildSections though